plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -284,6 +287,217 @@ public void testSelectivelyOptimizingLeavesEqualityDeletes()
+                .overwrite();
+        writer = writerBuilder.buildEqualityWriter();
+
+        dataDelete = GenericRecord.create(thirdDeleteRowSchema);
+        try (Closeable ignored = writer) {
+            writer.write(dataDelete.copy(ImmutableMap.of("regionkey", 2L)));
+        }
+        icebergTable.newRowDelta().addDeletes(writer.toDeleteFile()).commit();
+
+        assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE NOT ((regionkey = 1 AND name = 'BRAZIL') OR regionkey = 2 OR name = 'ALGERIA')");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testMultipleEqualityDeletesNestedFields()
+            throws Exception
+    {
+        String tableName = "test_equality_delets_different_schemas_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " ( id BIGINT, root ROW(nested BIGINT, nested_other BIGINT))");
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, row(10, 100))", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES (2, row(20, 200))", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES (2, row(20, 200))", 1);
+        Table icebergTable = loadTable(tableName);
+        assertThat(icebergTable.currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("0");
+        Path metadataDir = new Path(metastoreDir.toURI());
+        TrinoFileSystem fs = HDFS_FILE_SYSTEM_FACTORY.create(SESSION);
+
+        List<String> deleteFileColumns = ImmutableList.of("root.nested");
+        Schema deleteRowSchema = icebergTable.schema().select(deleteFileColumns);
+        List<Integer> equalityFieldIds = deleteFileColumns.stream()
+                .map(name -> deleteRowSchema.findField(name).fieldId())
+                .collect(toImmutableList());
+        for (int i = 1; i < 3; i++) {
+            String deleteFile = "delete_file_" + UUID.randomUUID();
+            Parquet.DeleteWriteBuilder writerBuilder = Parquet.writeDeletes(new ForwardingFileIo(fs).newOutputFile(new Path(metadataDir, deleteFile).toString()))
+                    .forTable(icebergTable)
+                    .rowSchema(deleteRowSchema)
+                    .createWriterFunc(GenericParquetWriter::buildWriter)
+                    .equalityFieldIds(equalityFieldIds)
+                    .overwrite();
+            EqualityDeleteWriter<Record> writer = writerBuilder.buildEqualityWriter();
+
+            Types.StructType nestedStructType = (Types.StructType) deleteRowSchema.findField("root").type();
+            Record dataDelete = GenericRecord.create(deleteRowSchema);
+            try (Closeable ignored = writer) {
+                Record nestedStruct = GenericRecord.create(nestedStructType);
+                nestedStruct.setField("nested", 20L);
+                writer.write(dataDelete.copy(ImmutableMap.of("root", nestedStruct)));
+            }
+            icebergTable.newRowDelta().addDeletes(writer.toDeleteFile()).commit();
```

Comment:
@Heltman Hi, I noticed that this file already contains the helper function `writeEqualityDeleteToNationTable` I think using it in your specs will shorten them significantly. just FYI in case you didn't notice the helper :)

Reply:
This has problem, `writeEqualityDeleteToNationTable` recovery schema by `overwriteValues`, but we need schema is `root.nested` and `overwriteValues` is `root`, so we can write `nested` success. Maybe we need `writeEqualityDeleteToNationTable` support `deleteRowSchema` param.

```java
Schema deleteRowSchema = icebergTable.schema().select(overwriteValues.keySet());
```

Reply:
Yea, that makes sense. It should be helpful as is for the non-nested tests already and adding a param + an override with a default should help with the nested test.

Reply:
I make it. Please check it! @jasonf20